waves-integration-it/src/test/scala/com/wavesplatform/dex/grpc/integration/clients/WavesBlockchainAsyncClientTestSuite.scala

```diff
@@ -0,0 +1,275 @@
+package com.wavesplatform.dex.grpc.integration.clients
+
+import java.nio.charset.StandardCharsets
+
+import com.wavesplatform.account.{Address, KeyPair}
+import com.wavesplatform.common.state.ByteStr
+import com.wavesplatform.common.utils.EitherExt2
+import com.wavesplatform.dex.grpc.integration.clients.WavesBlockchainClient.SpendableBalanceChanges
+import com.wavesplatform.dex.grpc.integration.dto.BriefAssetDescription
+import com.wavesplatform.dex.grpc.integration.{DEXClient, ItTestSuiteBase}
+import com.wavesplatform.features.BlockchainFeatures
+import com.wavesplatform.lang.script.v1.ExprScript
+import com.wavesplatform.lang.v1.compiler.Terms
+import com.wavesplatform.transaction.Asset
+import com.wavesplatform.transaction.Asset.{IssuedAsset, Waves}
+import com.wavesplatform.transaction.assets.exchange.{AssetPair, ExchangeTransactionV2, Order, OrderType}
+import monix.execution.Ack.Continue
+import monix.execution.{Ack, Scheduler}
+import monix.reactive.Observer
+import org.scalatest.{Assertion, BeforeAndAfterEach}
+
+import scala.concurrent.duration.DurationInt
+import scala.concurrent.{Await, ExecutionContext, Future}
+import scala.util.Random
+
+class WavesBlockchainAsyncClientTestSuite extends ItTestSuiteBase with BeforeAndAfterEach {
+
+  private val runNow = new ExecutionContext {
+    override def execute(runnable: Runnable): Unit     = runnable.run()
+    override def reportFailure(cause: Throwable): Unit = throw cause
+  }
+
+  private implicit val monixScheduler = Scheduler(runNow)
+  private lazy val client             = new DEXClient(wavesNode1GrpcApiTarget, 100.milliseconds).wavesBlockchainAsyncClient
+
+  override implicit def patienceConfig: PatienceConfig = super.patienceConfig.copy(
+    timeout = 1.minute,
+    interval = 1.second
+  )
+
+  @volatile private var balanceChanges = Map.empty[Address, Map[Asset, Long]]
+
+  private val eventsObserver: Observer[SpendableBalanceChanges] = new Observer[SpendableBalanceChanges] {
+    override def onError(ex: Throwable): Unit                       = Unit
+    override def onComplete(): Unit                                 = Unit
+    override def onNext(elem: SpendableBalanceChanges): Future[Ack] = { balanceChanges ++= elem; Continue }
+  }
+
+  private def assertBalanceChanges(expectedBalanceChanges: Map[Address, Map[Asset, Long]]): Assertion = eventually {
+    // Remove pairs (address, asset) those expectedBalanceChanges has not
+    val actual = simplify(balanceChanges.filterKeys(expectedBalanceChanges.keys.toSet).map {
+      case (address, balance) => address -> balance.filterKeys(expectedBalanceChanges(address).contains)
+    })
+    val expected = simplify(expectedBalanceChanges)
+    log.trace(s"Compare:\nactual: $actual\nexpected: $expected")
+    actual shouldBe expected
```

Comment:
what about diffx usage?

Reply:
Copied this code from other client and didn't think about it. Good idea :) I will